### PR TITLE
Fix reply to all, and some refactorings for initFromReplyTo

### LINKED ISSUE #1369: Answer acts like answer all
### Expected behaviour

Only the sender is copied to the reciever field on the compose screen.
### Actual behaviour

All the email addresses in the original email are copied to the reciever fields on the compose screen.
### Steps to reproduce
1. open a recieved mail with multiple recievers
2. Click on the answer icon button and then on answer in the popup
### Environment

K-9 Mail version: v5.108

Android version: 6.0.1; Security Patch: 1. March 2016)

Device: Nexus 5X

### PR DESCRIPTION
**Pull request by Valodim**
What the title says, fixes #1369. I also moved more "who to reply to" logic into ReplyToParser and added tests for that.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/helper/ReplyToParser.java [hunk 1]
```diff
@@ -1,23 +1,78 @@
...
+            result = message.getRecipients(RecipientType.TO);
         }
 
-        return message.getFrom();
+        return result;
     }
+
+    public static ReplyToAddresses getRecipientsToReplyAllTo(
+            Message message, Address[] replyToAddresses, Account account) {
+        ArrayList<Address> alreadyAddedAddresses = new ArrayList<>(Arrays.asList(replyToAddresses));
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/helper/ReplyToParser.java]: This should probably be a `HashSet`.

File: k9mail/src/main/java/com/fsck/k9/helper/ReplyToParser.java [hunk 2]
```diff
@@ -1,23 +1,78 @@
...
+        ArrayList<Address> toAddresses = new ArrayList<>();
+        ArrayList<Address> ccAddresses = new ArrayList<>();
+
+        for (Address address : message.getFrom()) {
+            if (!account.isAnIdentity(address) && !alreadyAddedAddresses.contains(address)) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/helper/ReplyToParser.java]: Switching the order of the conditions avoids calling the more expensive `isAnIdentity()` for addresses we have already added.

File: k9mail/src/main/java/com/fsck/k9/helper/ReplyToParser.java [hunk 3]
```diff
@@ -1,23 +1,78 @@
...
+        boolean replyToAddressIsUserIdentity = account.isAnIdentity(result);
+        if (replyToAddressIsUserIdentity) {
+            result = message.getRecipients(RecipientType.TO);
         }
 
-        return message.getFrom();
+        return result;
     }
+
+    public static ReplyToAddresses getRecipientsToReplyAllTo(
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/helper/ReplyToParser.java]: I'd prefer this to have the same signature as `getRecipientsToReplyTo()`. You'd then have to change the call site to only call one of the methods.

Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/helper/ReplyToParser.java]: makes sense

**Comment by Valodim** [comment]
done

**Pull request merged**